docs/design-proposal.md

```diff
@@ -0,0 +1,190 @@
+# Design Proposal: Isolated Job Runner
+
+Author(s): Dale Davis
+
+Last updated: 2024-04-30
+
+Discussion at [int-cloud-swe-dale](https://app.slack.com/client/TF04GMFKN/C070ZARGD6U) 
+Slack channel.
+
+<!-- TOC -->
+* [Design Proposal: Isolated Job Runner](#design-proposal-isolated-job-runner)
+  * [Abstract](#abstract)
+  * [Proposal](#proposal)
+    * [IJR Library](#ijr-library)
+      * [Technology Stack](#technology-stack)
+    * [ijrd](#ijrd)
+    * [ijrctl](#ijrctl)
+      * [ijrctl rootfs](#ijrctl-rootfs)
+      * [ijrctl job](#ijrctl-job)
+  * [Testing Strategy](#testing-strategy)
+  * [Milestones](#milestones)
+<!-- TOC -->
+
+## Abstract
+
+There is a need to execute jobs on Linux workers with process ID, mount, and
+networking isolation. Users need to define CPU, memory, and disk I/O limits for
+created jobs. Users can start and stop jobs, query job status, and stream the
+output of running jobs.
+
+## Proposal
+
+The solution will consist of one git repository for library package code, client
+and server CLI applications, unit tests, documentation, and build files. It 
+provides the ability to upload archive root file systems and to execute jobs
+consisting of commands or scripts located on the uploaded file systems. For this
+release, file system archives will be gzipped tarballs.
+
+Executing jobs cannot see the file system of the host the service is running on,
+or any of the other uploaded file systems other than the one its being executed
+on. Executing jobs have cannot access the host's network interfaces and have
+no network access of their own in this release.
+
+### IJR Library
```

Comment:
Can you provide more detail about the process execution lifecycle for jobs (e.g., how they are initialized with the required limits & isolation, how they are terminated, how child processes are managed, etc.)?

Reply:
Sure. I'll add a section describing the job process life cycle and repost.